symengine/type_codes.inc

```diff
@@ -1,3 +1,6 @@
+#ifdef NAN
+#undef NAN
+#endif
```

Comment:
This could create problems for other software including this header. I'd prefer if the typecode was changed.